### PR TITLE
Add Auth0 provider

### PR DESCRIPTION
**Pull request by rolodato**
Wanting to use Auth0 with bell so I implemented it as a provider. If this could eventually get merged I can write the tests and add it to docs.

Some references:

* https://auth0.com/docs/protocols#oauth-server-side
* https://auth0.com/docs/auth-api
* https://auth0.com/docs/user-profile/normalized

Thanks!

### CODE REVIEW

File: lib/providers/auth0.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+exports = module.exports = function (options) {
+
+    options = options || {};
```
Comment (ldesplat) [inline lib/providers/auth0.js]: You should be enforcing that `domain` exists and is a string. See phabricator.js for an example.

**Comment by ldesplat** [comment]
Add the docs and tests as well and I'll be very happy to accept this.

Thank you.

**Comment by rolodato** [comment]
@ldesplat Tests and docs added.

**Comment by ldesplat** [comment]
Thanks again :)

**Comment by rolodato** [comment]
No problem, thank you! Looking forward to the new release :+1:

**Comment by rolodato** [comment]
Thanks @travi. As of today Auth0 does not support configuring custom domains for login pages, but that's in our backlog (currently only supported for private instances which have their own domains instead of Auth0 subdomains).

Don't want to pollute this PR further, so if you want to discuss anything Auth0-related feel free to email me or create a thread at https;//ask.auth0.com. Thanks!

**Comment by rolodato** [comment]
For anyone stumbling upon this PR, I've created a Hapi + Auth0 sample here: https://github.com/auth0/node-auth0/tree/2a07a14de7add7188a2d997fe9f2af09173d912d/examples/hapi-regular-webapp

**Pull request merged**